src/hotspot/share/classfile/moduleEntry.cpp

```diff
@@ -208,7 +209,7 @@ void ModuleEntry::set_is_open(bool is_open) {
 // module will return false.
 bool ModuleEntry::has_reads_list() const {
   assert_locked_or_safepoint(Module_lock);
-  return ((_reads != nullptr) && !_reads->is_empty());
+  return ((_reads != nullptr) && !reads()->is_empty());
```

Comment:
Could the first condition be `reads() != nullptr` ?

Reply:
This was a vestige of the earlier version where the state booleans may not have been initialized. I believe it should ok to use `reads()` now.